{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### BLAST  on database search identifued peptides vs _de novo_ sequenced peptides ###"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mon Dec 12 13:18:31 PST 2016\r\n"
     ]
    }
   ],
   "source": [
    "!date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Software:\r\n",
      "\r\n",
      "    System Software Overview:\r\n",
      "\r\n",
      "      System Version: macOS 10.12.1 (16B2555)\r\n",
      "      Kernel Version: Darwin 16.1.0\r\n",
      "      Boot Volume: Macintosh HD\r\n",
      "      Boot Mode: Normal\r\n",
      "      Computer Name: Meganâ€™s MacBook Pro\r\n",
      "      User Name: Megan Duffy (meganduffy)\r\n",
      "      Secure Virtual Memory: Enabled\r\n",
      "      System Integrity Protection: Enabled\r\n",
      "      Time since boot: 28 days 4:21\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!system_profiler SPSoftwareDataType"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cd /Users/meganduffy/Documents/git-repos/FISH-546/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 12/12/2016 13:22:32\n",
      "New DB name:   /Users/meganduffy/Documents/git-repos/FISH-546/data/galaxy/062016_ETNP_129_100m_db_db.fasta\n",
      "New DB title:  /Users/meganduffy/Documents/git-repos/FISH-546/data/galaxy/062016_ETNP_129_100m_db.fasta\n",
      "Sequence type: Protein\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1229 sequences in 0.072823 seconds.\n"
     ]
    }
   ],
   "source": [
    "# making database\n",
    "!~/Applications/ncbi-blast-2.5.0+/bin/makeblastdb \\\n",
    "-in /Users/meganduffy/Documents/git-repos/FISH-546/data/galaxy/062016_ETNP_129_100m_db.fasta \\\n",
    "-dbtype prot \\\n",
    "-out /Users/meganduffy/Documents/git-repos/FISH-546/data/galaxy/062016_ETNP_129_100m_db_db.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# blastp search of de novo peptides vs database peptides\n",
    "# outfmt = 6 gives output in tabular format. \n",
    "!~/Applications/ncbi-blast-2.5.0+/bin/blastp \\\n",
    "-query /Users/meganduffy/Documents/git-repos/FISH-546/data/galaxy/062016_ETNP_129_100m_peaks.fasta \\\n",
    "-db /Users/meganduffy/Documents/git-repos/FISH-546/data/galaxy/062016_ETNP_129_100m_db_db.fasta \\\n",
    "-outfmt 10 \\\n",
    "-out /Users/meganduffy/Documents/git-repos/FISH-546/analyses/062016_ETNP_129_100m_peaks_v_db.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HSAPTDFLSLR\tCETACPTDFLSIR\t70.000\t10\t3\t0\t2\t11\t4\t13\t0.007\t20.4\r\n",
      "HSAPTDFLSLR\tCETACPTDFLSIR\t70.000\t10\t3\t0\t2\t11\t4\t13\t0.007\t20.4\r\n",
      "HSAPTDFLSLR\tCETACPTDFLSIR\t70.000\t10\t3\t0\t2\t11\t4\t13\t0.007\t20.4\r\n",
      "HSAPTDFLSLR\tRCETACPTDFLSIR\t70.000\t10\t3\t0\t2\t11\t5\t14\t0.007\t20.4\r\n",
      "HSAPTDFLSLR\tRCETACPTDFLSIR\t70.000\t10\t3\t0\t2\t11\t5\t14\t0.007\t20.4\r\n",
      "STAMVWQR\tTVVPNLSLAMVWQR\t87.500\t8\t1\t0\t1\t8\t7\t14\t0.034\t18.1\r\n",
      "STAMVWQR\tTVVPNLSLAMVWQR\t87.500\t8\t1\t0\t1\t8\t7\t14\t0.034\t18.1\r\n",
      "STAMVWQR\tTVVPNLSLAMVWQR\t87.500\t8\t1\t0\t1\t8\t7\t14\t0.034\t18.1\r\n",
      "STAMVWQR\tTVVPNLSLAMVWQR\t87.500\t8\t1\t0\t1\t8\t7\t14\t0.034\t18.1\r\n",
      "QPENLPLGPK\tTDENLPLGPK\t100.000\t8\t0\t0\t3\t10\t3\t10\t0.013\t19.6\r\n"
     ]
    }
   ],
   "source": [
    "! head /Users/meganduffy/Documents/git-repos/FISH-546/analyses/062016_ETNP_129_100m_peaks_v_db.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31mblast_formatter\u001b[m\u001b[m   \u001b[31mblastx\u001b[m\u001b[m            \u001b[31mmakembindex\u001b[m\u001b[m       \u001b[31mtblastn\u001b[m\u001b[m\r\n",
      "\u001b[31mblastdb_aliastool\u001b[m\u001b[m \u001b[31mconvert2blastmask\u001b[m\u001b[m \u001b[31mmakeprofiledb\u001b[m\u001b[m     \u001b[31mtblastx\u001b[m\u001b[m\r\n",
      "\u001b[31mblastdbcheck\u001b[m\u001b[m      \u001b[31mdeltablast\u001b[m\u001b[m        \u001b[31mpsiblast\u001b[m\u001b[m          \u001b[31mupdate_blastdb.pl\u001b[m\u001b[m\r\n",
      "\u001b[31mblastdbcmd\u001b[m\u001b[m        \u001b[31mdustmasker\u001b[m\u001b[m        \u001b[31mrpsblast\u001b[m\u001b[m          \u001b[31mwindowmasker\u001b[m\u001b[m\r\n",
      "\u001b[31mblastn\u001b[m\u001b[m            \u001b[31mlegacy_blast.pl\u001b[m\u001b[m   \u001b[31mrpstblastn\u001b[m\u001b[m\r\n",
      "\u001b[31mblastp\u001b[m\u001b[m            \u001b[31mmakeblastdb\u001b[m\u001b[m       \u001b[31msegmasker\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls ~/Applications/ncbi-blast-2.5.0+/bin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "USAGE\r\n",
      "  blastp [-h] [-help] [-import_search_strategy filename]\r\n",
      "    [-export_search_strategy filename] [-task task_name] [-db database_name]\r\n",
      "    [-dbsize num_letters] [-gilist filename] [-seqidlist filename]\r\n",
      "    [-negative_gilist filename] [-entrez_query entrez_query]\r\n",
      "    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\r\n",
      "    [-subject subject_input_file] [-subject_loc range] [-query input_file]\r\n",
      "    [-out output_file] [-evalue evalue] [-word_size int_value]\r\n",
      "    [-gapopen open_penalty] [-gapextend extend_penalty]\r\n",
      "    [-qcov_hsp_perc float_value] [-max_hsps int_value]\r\n",
      "    [-xdrop_ungap float_value] [-xdrop_gap float_value]\r\n",
      "    [-xdrop_gap_final float_value] [-searchsp int_value]\r\n",
      "    [-sum_stats bool_value] [-seg SEG_options] [-soft_masking soft_masking]\r\n",
      "    [-matrix matrix_name] [-threshold float_value] [-culling_limit int_value]\r\n",
      "    [-best_hit_overhang float_value] [-best_hit_score_edge float_value]\r\n",
      "    [-window_size int_value] [-lcase_masking] [-query_loc range]\r\n",
      "    [-parse_deflines] [-outfmt format] [-show_gis]\r\n",
      "    [-num_descriptions int_value] [-num_alignments int_value]\r\n",
      "    [-line_length line_length] [-html] [-max_target_seqs num_sequences]\r\n",
      "    [-num_threads int_value] [-ungapped] [-remote] [-comp_based_stats compo]\r\n",
      "    [-use_sw_tback] [-version]\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "   Protein-Protein BLAST 2.5.0+\r\n",
      "\r\n",
      "OPTIONAL ARGUMENTS\r\n",
      " -h\r\n",
      "   Print USAGE and DESCRIPTION;  ignore all other parameters\r\n",
      " -help\r\n",
      "   Print USAGE, DESCRIPTION and ARGUMENTS; ignore all other parameters\r\n",
      " -version\r\n",
      "   Print version number;  ignore other arguments\r\n",
      "\r\n",
      " *** Input query options\r\n",
      " -query <File_In>\r\n",
      "   Input file name\r\n",
      "   Default = `-'\r\n",
      " -query_loc <String>\r\n",
      "   Location on the query sequence in 1-based offsets (Format: start-stop)\r\n",
      "\r\n",
      " *** General search options\r\n",
      " -task <String, Permissible values: 'blastp' 'blastp-fast' 'blastp-short' >\r\n",
      "   Task to execute\r\n",
      "   Default = `blastp'\r\n",
      " -db <String>\r\n",
      "   BLAST database name\r\n",
      "    * Incompatible with:  subject, subject_loc\r\n",
      " -out <File_Out>\r\n",
      "   Output file name\r\n",
      "   Default = `-'\r\n",
      " -evalue <Real>\r\n",
      "   Expectation value (E) threshold for saving hits \r\n",
      "   Default = `10'\r\n",
      " -word_size <Integer, >=2>\r\n",
      "   Word size for wordfinder algorithm\r\n",
      " -gapopen <Integer>\r\n",
      "   Cost to open a gap\r\n",
      " -gapextend <Integer>\r\n",
      "   Cost to extend a gap\r\n",
      " -matrix <String>\r\n",
      "   Scoring matrix name (normally BLOSUM62)\r\n",
      " -threshold <Real, >=0>\r\n",
      "   Minimum word score such that the word is added to the BLAST lookup table\r\n",
      " -comp_based_stats <String>\r\n",
      "   Use composition-based statistics:\r\n",
      "       D or d: default (equivalent to 2 )\r\n",
      "       0 or F or f: No composition-based statistics\r\n",
      "       1: Composition-based statistics as in NAR 29:2994-3005, 2001\r\n",
      "       2 or T or t : Composition-based score adjustment as in Bioinformatics\r\n",
      "   21:902-911,\r\n",
      "       2005, conditioned on sequence properties\r\n",
      "       3: Composition-based score adjustment as in Bioinformatics 21:902-911,\r\n",
      "       2005, unconditionally\r\n",
      "   Default = `2'\r\n",
      "\r\n",
      " *** BLAST-2-Sequences options\r\n",
      " -subject <File_In>\r\n",
      "   Subject sequence(s) to search\r\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\r\n",
      "   db_soft_mask, db_hard_mask\r\n",
      " -subject_loc <String>\r\n",
      "   Location on the subject sequence in 1-based offsets (Format: start-stop)\r\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\r\n",
      "   db_soft_mask, db_hard_mask, remote\r\n",
      "\r\n",
      " *** Formatting options\r\n",
      " -outfmt <String>\r\n",
      "   alignment view options:\r\n",
      "     0 = Pairwise,\r\n",
      "     1 = Query-anchored showing identities,\r\n",
      "     2 = Query-anchored no identities,\r\n",
      "     3 = Flat query-anchored showing identities,\r\n",
      "     4 = Flat query-anchored no identities,\r\n",
      "     5 = BLAST XML,\r\n",
      "     6 = Tabular,\r\n",
      "     7 = Tabular with comment lines,\r\n",
      "     8 = Seqalign (Text ASN.1),\r\n",
      "     9 = Seqalign (Binary ASN.1),\r\n",
      "    10 = Comma-separated values,\r\n",
      "    11 = BLAST archive (ASN.1),\r\n",
      "    12 = Seqalign (JSON),\r\n",
      "    13 = Multiple-file BLAST JSON,\r\n",
      "    14 = Multiple-file BLAST XML2,\r\n",
      "    15 = Single-file BLAST JSON,\r\n",
      "    16 = Single-file BLAST XML2,\r\n",
      "    18 = Organism Report\r\n",
      "   \r\n",
      "   Options 6, 7 and 10 can be additionally configured to produce\r\n",
      "   a custom format specified by space delimited format specifiers.\r\n",
      "   The supported format specifiers are:\r\n",
      "   \t    qseqid means Query Seq-id\r\n",
      "   \t       qgi means Query GI\r\n",
      "   \t      qacc means Query accesion\r\n",
      "   \t   qaccver means Query accesion.version\r\n",
      "   \t      qlen means Query sequence length\r\n",
      "   \t    sseqid means Subject Seq-id\r\n",
      "   \t sallseqid means All subject Seq-id(s), separated by a ';'\r\n",
      "   \t       sgi means Subject GI\r\n",
      "   \t    sallgi means All subject GIs\r\n",
      "   \t      sacc means Subject accession\r\n",
      "   \t   saccver means Subject accession.version\r\n",
      "   \t   sallacc means All subject accessions\r\n",
      "   \t      slen means Subject sequence length\r\n",
      "   \t    qstart means Start of alignment in query\r\n",
      "   \t      qend means End of alignment in query\r\n",
      "   \t    sstart means Start of alignment in subject\r\n",
      "   \t      send means End of alignment in subject\r\n",
      "   \t      qseq means Aligned part of query sequence\r\n",
      "   \t      sseq means Aligned part of subject sequence\r\n",
      "   \t    evalue means Expect value\r",
      "\r\n",
      "   \t  bitscore means Bit score\r\n",
      "   \t     score means Raw score\r\n",
      "   \t    length means Alignment length\r\n",
      "   \t    pident means Percentage of identical matches\r\n",
      "   \t    nident means Number of identical matches\r\n",
      "   \t  mismatch means Number of mismatches\r\n",
      "   \t  positive means Number of positive-scoring matches\r\n",
      "   \t   gapopen means Number of gap openings\r\n",
      "   \t      gaps means Total number of gaps\r\n",
      "   \t      ppos means Percentage of positive-scoring matches\r\n",
      "   \t    frames means Query and subject frames separated by a '/'\r\n",
      "   \t    qframe means Query frame\r\n",
      "   \t    sframe means Subject frame\r\n",
      "   \t      btop means Blast traceback operations (BTOP)\r\n",
      "   \t    staxid means Subject Taxonomy ID\r\n",
      "   \t  ssciname means Subject Scientific Name\r\n",
      "   \t  scomname means Subject Common Name\r\n",
      "   \tsblastname means Subject Blast Name\r\n",
      "   \t sskingdom means Subject Super Kingdom\r\n",
      "   \t   staxids means unique Subject Taxonomy ID(s), separated by a ';'\r\n",
      "   \t\t\t (in numerical order)\r\n",
      "   \t sscinames means unique Subject Scientific Name(s), separated by a ';'\r\n",
      "   \t scomnames means unique Subject Common Name(s), separated by a ';'\r\n",
      "   \tsblastnames means unique Subject Blast Name(s), separated by a ';'\r\n",
      "   \t\t\t (in alphabetical order)\r\n",
      "   \tsskingdoms means unique Subject Super Kingdom(s), separated by a ';'\r\n",
      "   \t\t\t (in alphabetical order) \r\n",
      "   \t    stitle means Subject Title\r\n",
      "   \tsalltitles means All Subject Title(s), separated by a '<>'\r\n",
      "   \t   sstrand means Subject Strand\r\n",
      "   \t     qcovs means Query Coverage Per Subject\r\n",
      "   \t   qcovhsp means Query Coverage Per HSP\r\n",
      "   \t    qcovus means Query Coverage Per Unique Subject (blastn only)\r\n",
      "   When not provided, the default value is:\r\n",
      "   'qacc sacc pident length mismatch gapopen qstart qend sstart send evalue\r\n",
      "   bitscore', which is equivalent to the keyword 'std'\r\n",
      "   Default = `0'\r\n",
      " -show_gis\r\n",
      "   Show NCBI GIs in deflines?\r\n",
      " -num_descriptions <Integer, >=0>\r\n",
      "   Number of database sequences to show one-line descriptions for\r\n",
      "   Not applicable for outfmt > 4\r\n",
      "   Default = `500'\r\n",
      "    * Incompatible with:  max_target_seqs\r\n",
      " -num_alignments <Integer, >=0>\r\n",
      "   Number of database sequences to show alignments for\r\n",
      "   Default = `250'\r\n",
      "    * Incompatible with:  max_target_seqs\r\n",
      " -line_length <Integer, >=1>\r\n",
      "   Line length for formatting alignments\r\n",
      "   Not applicable for outfmt > 4\r\n",
      "   Default = `60'\r\n",
      " -html\r\n",
      "   Produce HTML output?\r\n",
      "\r\n",
      " *** Query filtering options\r\n",
      " -seg <String>\r\n",
      "   Filter query sequence with SEG (Format: 'yes', 'window locut hicut', or\r\n",
      "   'no' to disable)\r\n",
      "   Default = `no'\r\n",
      " -soft_masking <Boolean>\r\n",
      "   Apply filtering locations as soft masks\r\n",
      "   Default = `false'\r\n",
      " -lcase_masking\r\n",
      "   Use lower case filtering in query and subject sequence(s)?\r\n",
      "\r\n",
      " *** Restrict search or results\r\n",
      " -gilist <String>\r\n",
      "   Restrict search of database to list of GI's\r\n",
      "    * Incompatible with:  negative_gilist, seqidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -seqidlist <String>\r\n",
      "   Restrict search of database to list of SeqId's\r\n",
      "    * Incompatible with:  gilist, negative_gilist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -negative_gilist <String>\r\n",
      "   Restrict search of database to everything except the listed GIs\r\n",
      "    * Incompatible with:  gilist, seqidlist, remote, subject, subject_loc\r\n",
      " -entrez_query <String>\r\n",
      "   Restrict search with the given Entrez query\r\n",
      "    * Requires:  remote\r\n",
      " -db_soft_mask <String>\r\n",
      "   Filtering algorithm ID to apply to the BLAST database as soft masking\r\n",
      "    * Incompatible with:  db_hard_mask, subject, subject_loc\r\n",
      " -db_hard_mask <String>\r\n",
      "   Filtering algorithm ID to apply to the BLAST database as hard masking\r\n",
      "    * Incompatible with:  db_soft_mask, subject, subject_loc\r\n",
      " -qcov_hsp_perc <Real, 0..100>\r\n",
      "   Percent query coverage per hsp\r\n",
      " -max_hsps <Integer, >=1>\r\n",
      "   Set maximum number of HSPs per subject sequence to save for each query\r\n",
      " -culling_limit <Integer, >=0>\r\n",
      "   If the query range of a hit is enveloped by that of at least this many\r\n",
      "   higher-scoring hits, delete the hit\r\n",
      "    * Incompatible with:  best_hit_overhang, best_hit_score_edge\r\n",
      " -best_hit_overhang <Real, (>0 and <0.5)>\r\n",
      "   Best Hit algorithm overhang value (recommended value: 0.1)\r\n",
      "    * Incompatible with:  culling_limit\r\n",
      " -best_hit_score_edge <Real, (>0 and <0.5)>\r\n",
      "   Best Hit algorithm score edge value (recommended value: 0.1)\r\n",
      "    * Incompatible with:  culling_limit\r\n",
      " -max_target_seqs <Integer, >=1>\r\n",
      "   Maximum number of aligned sequences to keep \r\n",
      "   Not applicable for outfmt <= 4\r\n",
      "   Default = `500'\r\n",
      "    * Incompatible with:  num_descriptions, num_alignments\r\n",
      "\r\n",
      " *** Statistical options\r\n",
      " -dbsize <Int8>\r\n",
      "   Effective length of the database \r\n",
      " -searchsp <Int8, >=0>\r\n",
      "   Effective length of the search space\r\n",
      " -sum_stats <Boolean>\r\n",
      "   Use sum statistics\r\n",
      "\r\n",
      " *** Search strategy options\r\n",
      " -import_search_strategy <File_In>\r\n",
      "   Search strategy to use\r\n",
      "    * Incompatible with:  export_search_strategy\r\n",
      " -export_search_strategy <File_Out>\r\n",
      "   File name to record the search strategy used\r\n",
      "    * Incompatible with:  import_search_strategy\r\n",
      "\r\n",
      " *** Extension options\r\n",
      " -xdrop_ungap <Real>\r\n",
      "   X-dropoff value (in bits) for ungapped extensions\r\n",
      " -xdrop_gap <Real>\r\n",
      "   X-dropoff value (in bits) for preliminary gapped extensions\r\n",
      " -xdrop_gap_final <Real>\r\n",
      "   X-dropoff value (in bits) for final gapped alignment\r\n",
      " -window_size <Integer, >=0>\r\n",
      "   Multiple hits window size, use 0 to specify 1-hit algorithm\r\n",
      " -ungapped\r\n",
      "   Perform ungapped alignment only?\r\n",
      "\r\n",
      " *** Miscellaneous options\r\n",
      " -parse_deflines\r\n",
      "   Should the query and subject defline(s) be parsed?\r\n",
      " -num_threads <Integer, >=1>\r\n",
      "   Number of threads (CPUs) to use in the BLAST search\r\n",
      "   Default = `1'\r\n",
      "    * Incompatible with:  remote\r\n",
      " -remote\r\n",
      "   Execute search remotely?\r\n",
      "    * Incompatible with:  gilist, seqidlist, negative_gilist, subject_loc,\r\n",
      "   num_threads\r\n",
      " -use_sw_tback\r\n",
      "   Compute locally optimal Smith-Waterman alignments?\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "! ~/Applications/ncbi-blast-2.5.0+/bin/blastp -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HSAPTDFLSLR,CETACPTDFLSIR,70.000,10,3,0,2,11,4,13,0.007,20.4\r\n",
      "HSAPTDFLSLR,CETACPTDFLSIR,70.000,10,3,0,2,11,4,13,0.007,20.4\r\n",
      "HSAPTDFLSLR,CETACPTDFLSIR,70.000,10,3,0,2,11,4,13,0.007,20.4\r\n",
      "HSAPTDFLSLR,RCETACPTDFLSIR,70.000,10,3,0,2,11,5,14,0.007,20.4\r\n",
      "HSAPTDFLSLR,RCETACPTDFLSIR,70.000,10,3,0,2,11,5,14,0.007,20.4\r\n",
      "STAMVWQR,TVVPNLSLAMVWQR,87.500,8,1,0,1,8,7,14,0.034,18.1\r\n",
      "STAMVWQR,TVVPNLSLAMVWQR,87.500,8,1,0,1,8,7,14,0.034,18.1\r\n",
      "STAMVWQR,TVVPNLSLAMVWQR,87.500,8,1,0,1,8,7,14,0.034,18.1\r\n",
      "STAMVWQR,TVVPNLSLAMVWQR,87.500,8,1,0,1,8,7,14,0.034,18.1\r\n",
      "QPENLPLGPK,TDENLPLGPK,100.000,8,0,0,3,10,3,10,0.013,19.6\r\n",
      "QPENLPLGPK,TDENLPLGPK,100.000,8,0,0,3,10,3,10,0.013,19.6\r\n",
      "QPENLPLGPK,TDENLPLGPK,100.000,8,0,0,3,10,3,10,0.013,19.6\r\n",
      "QPENLPLGPK,TDENLPLGPK,100.000,8,0,0,3,10,3,10,0.013,19.6\r\n",
      "QPENLPLGPK,ENLPLGPK,100.000,8,0,0,3,10,1,8,0.016,19.2\r\n",
      "QPENLPLGPK,ENLPLGPK,100.000,8,0,0,3,10,1,8,0.016,19.2\r\n",
      "QPENLPLGPK,NLPLGPK,100.000,7,0,0,4,10,1,7,0.084,17.3\r\n",
      "QPENLPLGPK,NLPLGPK,100.000,7,0,0,4,10,1,7,0.084,17.3\r\n",
      "QPENLPLGPK,NLPLGPK,100.000,7,0,0,4,10,1,7,0.084,17.3\r\n",
      "QPENLPLGPK,NLPLGPK,100.000,7,0,0,4,10,1,7,0.084,17.3\r\n",
      "AAPFETFYTK,AAEDPEFETFYTK,100.000,7,0,0,4,10,7,13,0.059,17.7\r\n",
      "AAPFETFYTK,AAEDPEFETFYTK,100.000,7,0,0,4,10,7,13,0.059,17.7\r\n",
      "AAPFETFYTK,AAEDPEFETFYTK,100.000,7,0,0,4,10,7,13,0.059,17.7\r\n",
      "AAPFETFYTK,AAEDPEFETFYTK,100.000,7,0,0,4,10,7,13,0.059,17.7\r\n",
      "GPDLTAWRR,AALDGADLTAVVLDHADLSR,83.333,6,1,0,1,6,5,10,3.8,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPARK,100.000,6,0,0,5,10,9,14,4.4,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPARK,100.000,6,0,0,5,10,9,14,4.4,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPARK,100.000,6,0,0,5,10,9,14,4.4,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPARK,100.000,6,0,0,5,10,9,14,4.4,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPAR,100.000,6,0,0,5,10,9,14,4.8,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPAR,100.000,6,0,0,5,10,9,14,4.8,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPAR,100.000,6,0,0,5,10,9,14,4.8,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPAR,100.000,6,0,0,5,10,9,14,4.8,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPAR,100.000,6,0,0,5,10,9,14,4.8,13.9\r\n",
      "AAHGNTTSAAVPK,AASIAMITNTTSAAPAR,100.000,6,0,0,5,10,9,14,4.8,13.9\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPARK,75.000,8,2,0,1,8,7,14,1.8,14.6\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPARK,75.000,8,2,0,1,8,7,14,1.8,14.6\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPARK,75.000,8,2,0,1,8,7,14,1.8,14.6\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPARK,75.000,8,2,0,1,8,7,14,1.8,14.6\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPAR,75.000,8,2,0,1,8,7,14,2.0,14.2\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPAR,75.000,8,2,0,1,8,7,14,2.0,14.2\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPAR,75.000,8,2,0,1,8,7,14,2.0,14.2\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPAR,75.000,8,2,0,1,8,7,14,2.0,14.2\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPAR,75.000,8,2,0,1,8,7,14,2.0,14.2\r\n",
      "LYNTTSAAVPK,AASIAMITNTTSAAPAR,75.000,8,2,0,1,8,7,14,2.0,14.2\r\n",
      "TSAGEYMAFK,QAGEYTTFK,66.667,9,3,0,2,10,1,9,0.17,16.5\r\n",
      "TSAGEYMAFK,QAGEYTTFK,66.667,9,3,0,2,10,1,9,0.17,16.5\r\n",
      "TSAGEYMAFK,QAGEYTTFK,66.667,9,3,0,2,10,1,9,0.17,16.5\r\n",
      "TSAGEYMAFK,QAGEYTTFK,66.667,9,3,0,2,10,1,9,0.17,16.5\r\n",
      "TSAGEYMAFK,QAGEYTTFK,66.667,9,3,0,2,10,1,9,0.17,16.5\r\n",
      "TSAGEYMAFK,AGEYTTFK,75.000,8,2,0,3,10,1,8,0.19,16.5\r\n",
      "HSAEPNLPLYR,ALVNNLPLYR,100.000,6,0,0,6,11,5,10,0.18,16.5\r\n",
      "HSAEPNLPLYR,ALVNNLPLYR,100.000,6,0,0,6,11,5,10,0.18,16.5\r\n",
      "HSAEPNLPLYR,ALVNNLPLYR,100.000,6,0,0,6,11,5,10,0.18,16.5\r\n",
      "HSAEPNLPLYR,ALVNNLPLYR,100.000,6,0,0,6,11,5,10,0.18,16.5\r\n",
      "HSAEPNLPLYR,VNNLPLYR,100.000,6,0,0,6,11,3,8,0.20,16.5\r\n",
      "HSAEPNLPLYR,NLPLYR,100.000,6,0,0,6,11,1,6,0.23,16.5\r\n",
      "NDGSANCPTDFLSLR,CETACPTDFLSIR,88.889,9,1,0,7,15,5,13,0.002,22.3\r\n",
      "NDGSANCPTDFLSLR,CETACPTDFLSIR,88.889,9,1,0,7,15,5,13,0.002,22.3\r\n",
      "NDGSANCPTDFLSLR,CETACPTDFLSIR,88.889,9,1,0,7,15,5,13,0.002,22.3\r\n",
      "NDGSANCPTDFLSLR,RCETACPTDFLSIR,88.889,9,1,0,7,15,6,14,0.002,22.3\r\n",
      "NDGSANCPTDFLSLR,RCETACPTDFLSIR,88.889,9,1,0,7,15,6,14,0.002,22.3\r\n",
      "NDGSANCPTDFLSLR,LGANIGSAMGPTGLGK,54.545,11,5,0,1,11,4,14,1.8,14.6\r\n",
      "NDGSANCPTDFLSLR,LGANIGSAMGPTGLGK,54.545,11,5,0,1,11,4,14,1.8,14.6\r\n",
      "NDGSANCPTDFLSLR,LGANIGSAMGPTGLGK,54.545,11,5,0,1,11,4,14,1.8,14.6\r\n",
      "NDGSANCPTDFLSLR,LGANIGSAMGPTGLGK,54.545,11,5,0,1,11,4,14,1.8,14.6\r\n",
      "NDGSANCPTDFLSLR,VVTPLGQPMDGKGEIPTTENR,50.000,8,4,0,2,9,10,17,6.7,13.9\r\n",
      "NDGSANCPTDFLSLR,VVTPLGQPMDGKGEIPTTENR,50.000,8,4,0,2,9,10,17,6.7,13.9\r",
      "\r\n",
      "NDGSANCPTDFLSLR,VVTPLGQPMDGKGEIPTTENR,50.000,8,4,0,2,9,10,17,6.7,13.9\r\n",
      "NDGSANCPTDFLSLR,VVTPLGQPMDGKGEIPTTENR,50.000,8,4,0,2,9,10,17,6.7,13.9\r\n",
      "NDGSANCPTDFLSLR,VVTPLGQPMDGKGEIPTTENR,50.000,8,4,0,2,9,10,17,6.7,13.9\r\n",
      "HVVCGGCTKCVR,IYDTCIGCTQCVR,58.333,12,5,0,1,12,2,13,0.054,18.1\r\n",
      "HVVCGGCTKCVR,IYDTCIGCTQCVR,58.333,12,5,0,1,12,2,13,0.054,18.1\r\n",
      "HVVCGGCTKCVR,IYDTCIGCTQCVR,58.333,12,5,0,1,12,2,13,0.054,18.1\r\n",
      "HVVCGGCTKCVR,GCTQCVR,85.714,7,1,0,6,12,1,7,0.15,16.9\r\n",
      "HVVCGGCTKCVR,GCTQCVR,85.714,7,1,0,6,12,1,7,0.15,16.9\r\n",
      "HVVCGGCTKCVR,GCTQCVR,85.714,7,1,0,6,12,1,7,0.15,16.9\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPARK,63.636,11,4,0,1,11,7,17,0.035,18.5\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPARK,63.636,11,4,0,1,11,7,17,0.035,18.5\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPARK,63.636,11,4,0,1,11,7,17,0.035,18.5\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPARK,63.636,11,4,0,1,11,7,17,0.035,18.5\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPAR,63.636,11,4,0,1,11,7,17,0.037,18.5\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPAR,63.636,11,4,0,1,11,7,17,0.037,18.5\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPAR,63.636,11,4,0,1,11,7,17,0.037,18.5\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPAR,63.636,11,4,0,1,11,7,17,0.037,18.5\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPAR,63.636,11,4,0,1,11,7,17,0.037,18.5\r\n",
      "VGNSDSAAPAR,AASIAMITNTTSAAPAR,63.636,11,4,0,1,11,7,17,0.037,18.5\r\n",
      "VGNSDSAAPAR,SAAPAR,100.000,6,0,0,6,11,1,6,2.7,14.2\r\n",
      "VGNSDSAAPAR,SAAPAR,100.000,6,0,0,6,11,1,6,2.7,14.2\r\n",
      "VGNSDSAAPAR,RVTNAIGNADNATITLN,57.143,7,3,0,1,7,6,12,3.0,14.2\r\n",
      "VGNSDSAAPAR,VTNAIGNADNATITLN,57.143,7,3,0,1,7,5,11,3.2,14.2\r\n",
      "VGNSDSAAPAR,VTNAIGNADNATITLN,57.143,7,3,0,1,7,5,11,3.2,14.2\r\n",
      "VGNSDSAAPAR,VTNAIGNADNATITLN,57.143,7,3,0,1,7,5,11,3.2,14.2\r\n",
      "VGNSDSAAPAR,VTNAIGNADNATITLN,57.143,7,3,0,1,7,5,11,3.2,14.2\r\n",
      "ENCCATDENLPLGPK,TDENLPLGPK,100.000,10,0,0,6,15,1,10,3.52e-04,24.3\r\n",
      "ENCCATDENLPLGPK,TDENLPLGPK,100.000,10,0,0,6,15,1,10,3.52e-04,24.3\r\n",
      "ENCCATDENLPLGPK,TDENLPLGPK,100.000,10,0,0,6,15,1,10,3.52e-04,24.3\r\n",
      "ENCCATDENLPLGPK,TDENLPLGPK,100.000,10,0,0,6,15,1,10,3.52e-04,24.3\r\n",
      "ENCCATDENLPLGPK,ENLPLGPK,100.000,8,0,0,8,15,1,8,0.014,20.0\r\n",
      "ENCCATDENLPLGPK,ENLPLGPK,100.000,8,0,0,8,15,1,8,0.014,20.0\r\n",
      "ENCCATDENLPLGPK,NLPLGPK,100.000,7,0,0,9,15,1,7,0.074,18.1\r\n",
      "ENCCATDENLPLGPK,NLPLGPK,100.000,7,0,0,9,15,1,7,0.074,18.1\r\n",
      "ENCCATDENLPLGPK,NLPLGPK,100.000,7,0,0,9,15,1,7,0.074,18.1\r\n",
      "ENCCATDENLPLGPK,NLPLGPK,100.000,7,0,0,9,15,1,7,0.074,18.1\r\n",
      "EDELCDGPTNK,DFGYSFPCDGPGRGGTCDISAWDAFYLAVFWALNTIAWTQYYWHWK,100.000,4,0,0,5,8,8,11,0.36,16.2\r\n",
      "EDELCDGPTNK,FSCDGPGR,100.000,4,0,0,5,8,3,6,3.5,14.2\r\n",
      "EDELCDGPTNK,FSCDGPGR,100.000,4,0,0,5,8,3,6,3.5,14.2\r\n",
      "EDELCDGPTNK,DFGYSFPCDGPGR,100.000,4,0,0,5,8,8,11,4.0,13.9\r\n",
      "EDELCDGPTNK,DFGYSFPCDGPGR,100.000,4,0,0,5,8,8,11,4.0,13.9\r\n",
      "EDELCDGPTNK,DFGYSFPCDGPGR,100.000,4,0,0,5,8,8,11,4.0,13.9\r\n",
      "EDELCDGPTNK,DFGYSFPCDGPGR,100.000,4,0,0,5,8,8,11,4.0,13.9\r\n",
      "EDELCDGPTNK,TCGGPPNGIQVER,66.667,6,2,0,5,10,2,7,6.0,13.5\r\n",
      "EDELCDGPTNK,TCGGPPNGIQVER,66.667,6,2,0,5,10,2,7,6.0,13.5\r\n",
      "EDELCDGPTNK,TCGGPPNGIQVER,66.667,6,2,0,5,10,2,7,6.0,13.5\r\n",
      "EDELCDGPTNK,PCDGPGR,100.000,4,0,0,5,8,2,5,6.1,13.9\r\n",
      "EDELCDGPTNK,PCDGPGR,100.000,4,0,0,5,8,2,5,6.1,13.9\r\n",
      "EDELCDGPTNK,PCDGPGR,100.000,4,0,0,5,8,2,5,6.1,13.9\r\n",
      "EDELCDGPTNK,TCGGPPNGIQVERDR,66.667,6,2,0,5,10,2,7,6.7,13.5\r\n",
      "EDELCDGPTNK,TCGGPPNGIQVERDR,66.667,6,2,0,5,10,2,7,6.7,13.5\r\n",
      "LAPMGETFYTK,AAEDPEFETFYTK,77.778,9,2,0,3,11,5,13,0.10,17.3\r\n",
      "LAPMGETFYTK,AAEDPEFETFYTK,77.778,9,2,0,3,11,5,13,0.10,17.3\r\n",
      "LAPMGETFYTK,AAEDPEFETFYTK,77.778,9,2,0,3,11,5,13,0.10,17.3\r\n",
      "LAPMGETFYTK,AAEDPEFETFYTK,77.778,9,2,0,3,11,5,13,0.10,17.3\r\n",
      "ESTNGQTDENLPLGPK,TDENLPLGPK,100.000,10,0,0,7,16,1,10,4.15e-04,23.9\r\n",
      "ESTNGQTDENLPLGPK,TDENLPLGPK,100.000,10,0,0,7,16,1,10,4.15e-04,23.9\r\n",
      "ESTNGQTDENLPLGPK,TDENLPLGPK,100.000,10,0,0,7,16,1,10,4.15e-04,23.9\r\n",
      "ESTNGQTDENLPLGPK,TDENLPLGPK,100.000,10,0,0,7,16,1,10,4.15e-04,23.9\r\n",
      "ESTNGQTDENLPLGPK,ENLPLGPK,100.000,8,0,0,9,16,1,8,0.016,20.0\r\n",
      "ESTNGQTDENLPLGPK,ENLPLGPK,100.000,8,0,0,9,16,1,8,0.016,20.0\r\n",
      "ESTNGQTDENLPLGPK,NLPLGPK,100.000,7,0,0,10,16,1,7,0.079,18.1\r\n",
      "ESTNGQTDENLPLGPK,NLPLGPK,100.000,7,0,0,10,16,1,7,0.079,18.1\r\n",
      "ESTNGQTDENLPLGPK,NLPLGPK,100.000,7,0,0,10,16,1,7,0.079,18.1\r\n",
      "ESTNGQTDENLPLGPK,NLPLGPK,100.000,7,0,0,10,16,1,7,0.079,18.1\r\n",
      "HGSPLAATLAAPK,VGDPIAALDGAPK,58.333,12,5,0,2,13,2,13,0.17,16.5\r\n",
      "HGSPLAATLAAPK,VGDPIAALDGAPK,58.333,12,5,0,2,13,2,13,0.17,16.5\r\n",
      "LNAPGGLLLMAGPR,MAIEEINAAGGVEVDGK,57.143,7,3,0,1,7,6,12,6.3,13.9\r\n",
      "RGHVNLLFPPR,AWMSTADQPHENFVFPEEVMPRGNAL,57.143,7,3,0,3,9,10,16,2.8,14.2\r\n",
      "DPPCLGGLCTNCR,DFGYSFPCDGPGRGGTCDISAWDAFYLAVFWALNTIAWTQYYWHWK,55.556,9,4,0,1,9,9,17,2.7,14.2\r\n",
      "SSPQLNDMER,AIEGMQPQINDMER,87.500,8,1,0,3,10,7,14,0.007,20.4\r\n",
      "SSPQLNDMER,AIEGMQPQINDMER,87.500,8,1,0,3,10,7,14,0.007,20.4\r\n",
      "SSPQLNDMER,AIEGMQPQINDMER,87.500,8,1,0,3,10,7,14,0.007,20.4\r\n",
      "SSPQLNDMER,AIEGMQPQINDMER,87.500,8,1,0,3,10,7,14,0.007,20.4\r\n",
      "VAPTLAPR,LAPTIQPR,62.500,8,3,0,1,8,1,8,0.57,15.0\r\n",
      "VAPTLAPR,LAPTIQPR,62.500,8,3,0,1,8,1,8,0.57,15.0\r\n",
      "VAPTLAPR,LAPTIQPR,62.500,8,3,0,1,8,1,8,0.57,15.0\r\n",
      "VAPTLAPR,LAPTIQPR,62.500,8,3,0,1,8,1,8,0.57,15.0\r\n",
      "VAPTLAPR,PTIQPR,66.667,6,2,0,3,8,1,6,8.8,13.5\r\n",
      "VAPTLAPR,PTIQPR,66.667,6,2,0,3,8,1,6,8.8,13.5\r\n",
      "VAPTLAPR,PTIQPR,66.667,6,2,0,3,8,1,6,8.8,13.5\r\n",
      "VAPTLAPR,PTIQPR,66.667,6,2,0,3,8,1,6,8.8,13.5\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTK,69.231,13,4,0,4,16,2,14,0.001,22.3\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTK,69.231,13,4,0,4,16,2,14,0.001,22.3\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTK,69.231,13,4,0,4,16,2,14,0.001,22.3\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTK,69.231,13,4,0,4,16,2,14,0.001,22.3\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTK,69.231,13,4,0,4,16,2,14,0.001,22.3\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTK,69.231,13,4,0,4,16,2,14,0.001,22.3\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTK,69.231,13,4,0,4,16,2,14,0.001,22.3\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTK,69.231,13,4,0,4,16,2,14,0.001,22.3\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTKAAEVR,69.231,13,4,0,4,16,2,14,0.002,22.3\r\n",
      "AVVMVPALEAADACTR,GLVPAIEAADAMTKAAEVR,69.231,13,4,0,4,16,2,14,0.002,22.3\r\n",
      "WVVPGGLSLAMVWQR,TVVPNLSLAMVWQR,85.714,14,1,1,2,15,2,14,0.001,22.7\r\n",
      "WVVPGGLSLAMVWQR,TVVPNLSLAMVWQR,85.714,14,1,1,2,15,2,14,0.001,22.7\r\n",
      "WVVPGGLSLAMVWQR,TVVPNLSLAMVWQR,85.714,14,1,1,2,15,2,14,0.001,22.7\r\n",
      "WVVPGGLSLAMVWQR,TVVPNLSLAMVWQR,85.714,14,1,1,2,15,2,14,0.001,22.7\r\n",
      "WVVPGGLSLAMVWQR,GPNGLSLEK,83.333,6,1,0,4,9,2,7,7.4,13.9\r\n",
      "WVVPGGLSLAMVWQR,GPNGLSLEK,83.333,6,1,0,4,9,2,7,7.4,13.9\r\n",
      "WVVPGGLSLAMVWQR,GPNGLSLEK,83.333,6,1,0,4,9,2,7,7.4,13.9\r\n",
      "LPATWLGFSPANPR,VSILGFGSFEPR,41.667,12,7,0,3,14,1,12,2.0,14.6\r\n",
      "LPATWLGFSPANPR,VSILGFGSFEPR,41.667,12,7,0,3,14,1,12,2.0,14.6\r\n",
      "LPATWLGFSPANPR,VSILGFGSFEPR,41.667,12,7,0,3,14,1,12,2.0,14.6\r\n",
      "LPATWLGFSPANPR,VSILGFGSFEPR,41.667,12,7,0,3,14,1,12,2.0,14.6\r\n",
      "ALPLLDYVGNSPAK,LAFYDYVGNSPAK,76.923,13,3,0,2,14,1,13,2.72e-04,24.3\r\n",
      "ALPLLDYVGNSPAK,LAFYDYVGNSPAK,76.923,13,3,0,2,14,1,13,2.72e-04,24.3\r\n",
      "ALPLLDYVGNSPAK,LAFYDYVGNSPAK,76.923,13,3,0,2,14,1,13,2.72e-04,24.3\r\n",
      "RPHDFVSTSPVK,LAFYDYVGNSPAK,50.000,10,5,0,3,12,4,13,0.15,16.9\r\n",
      "RPHDFVSTSPVK,LAFYDYVGNSPAK,50.000,10,5,0,3,12,4,13,0.15,16.9\r\n",
      "RPHDFVSTSPVK,LAFYDYVGNSPAK,50.000,10,5,0,3,12,4,13,0.15,16.9\r\n",
      "RPHDFVSTSPVK,RCETACPTDFLSIR,66.667,6,2,0,2,7,7,12,8.0,13.5\r\n",
      "RPHDFVSTSPVK,RCETACPTDFLSIR,66.667,6,2,0,2,7,7,12,8.0,13.5\r\n",
      "RPHDFVSTSPVK,CETACPTDFLSIR,66.667,6,2,0,2,7,6,11,8.5,13.5\r\n",
      "RPHDFVSTSPVK,CETACPTDFLSIR,66.667,6,2,0,2,7,6,11,8.5,13.5\r\n",
      "RPHDFVSTSPVK,CETACPTDFLSIR,66.667,6,2,0,2,7,6,11,8.5,13.5\r\n",
      "APAGNELLVYMR,VITETIPAGPSLLVK,75.000,8,2,0,2,9,7,14,1.00,15.0\r\n",
      "APAGNELLVYMR,VITETIPAGPSLLVK,75.000,8,2,0,2,9,7,14,1.00,15.0\r\n",
      "APAGNELLVYMR,VITETIPAGPSLLVK,75.000,8,2,0,2,9,7,14,1.00,15.0\r\n",
      "CKFMLDAAYR,VGLLGDDFVLNPSYR,50.000,8,4,0,3,10,8,15,1.5,14.6\r\n",
      "CKFMLDAAYR,VGLLGDDFVLNPSYR,50.000,8,4,0,3,10,8,15,1.5,14.6\r\n",
      "WTDPPK,GPTYTDPAK,66.667,6,2,0,1,6,4,9,8.9,13.1\r\n",
      "NPSPAAVYDSAERVR,YYGLYDSAEGVVR,66.667,9,3,0,6,14,3,11,0.38,15.8\r\n",
      "NPSPAAVYDSAERVR,YYGLYDSAEGVVR,66.667,9,3,0,6,14,3,11,0.38,15.8\r\n",
      "NPSPAAVYDSAERVR,YYGLYDSAEGVVR,66.667,9,3,0,6,14,3,11,0.38,15.8\r\n",
      "NPSPAAVYDSAERVR,IGENPNPASIKFSGR,45.455,11,6,0,1,11,4,14,1.2,15.0\r\n",
      "NPSPAAVYDSAERVR,IGENPNPASIK,57.143,7,3,0,1,7,4,10,1.9,14.6\r\n",
      "NPSPAAVYDSAERVR,IGENPNPAGLKFTTK,57.143,7,3,0,1,7,4,10,2.7,14.2\r\n",
      "NPSPAAVYDSAERVR,EGYFPVAPNDTAQDIR,41.667,12,7,0,4,15,5,16,5.3,13.9\r\n",
      "GGSSAPLAALTAAPK,VGDPIAALDGAPK,70.000,10,3,0,6,15,4,13,0.12,17.3\r\n",
      "GGSSAPLAALTAAPK,VGDPIAALDGAPK,70.000,10,3,0,6,15,4,13,0.12,17.3\r\n",
      "GGSSAPLAALTAAPK,TMVLAMTAAPK,85.714,7,1,0,9,15,5,11,0.68,15.4\r\n",
      "QTPAGGMLAASGLEEK,YLGSTGGLLNSAETEEK,46.154,13,7,0,4,16,5,17,1.6,15.0\r\n",
      "QTPAGGMLAASGLEEK,YLGSTGGLLNSAETEEK,46.154,13,7,0,4,16,5,17,1.6,15.0\r\n",
      "QTPAGGMLAASGLEEK,YLGSTGGLLNSAETEEK,46.154,13,7,0,4,16,5,17,1.6,15.0\r\n",
      "DRPYGAPGK,GLSFGAPGSAGIGK,66.667,6,2,0,4,9,4,9,6.8,13.5\r\n",
      "DRPYGAPGK,GLSFGAPGSAGIGK,66.667,6,2,0,4,9,4,9,6.8,13.5\r\n",
      "DRPYGAPGK,GLSFGAPGSAGIGK,66.667,6,2,0,4,9,4,9,6.8,13.5\r\n",
      "SVDDPGYSLLEACMHK,DWVSNMEDTHYALGTACGPHPFPEIVSYFQSIVGKESREQLLANGAK,38.462,13,8,0,1,13,5,17,3.7,14.2\r\n",
      "NSGELLAPER,LQEILAPEIIVR,83.333,6,1,0,4,9,3,8,1.6,14.6\r\n",
      "DYGQGVGAVCAPK,ALTTGVDYAQGLVALGGDDK,55.556,9,4,0,1,9,7,15,4.4,13.9\r\n",
      "DYGQGVGAVCAPK,ALTTGVDYAQGLVALGGDDK,55.556,9,4,0,1,9,7,15,4.4,13.9\r\n",
      "DYGQGVGAVCAPK,ALTTGVDYAQGLVALGGDDK,55.556,9,4,0,1,9,7,15,4.4,13.9\r\n",
      "DYGQGVGAVCAPK,ALTTGVDYAQGLVALGGDDK,55.556,9,4,0,1,9,7,15,4.4,13.9\r\n",
      "RPLNATSEPK,GANTGIEVVGINVTSDPK,62.500,8,3,0,3,10,11,18,0.21,16.2\r\n",
      "KAGGFGEPGNPAAR,ETFGALGVPTGSAAR,50.000,12,6,0,3,14,4,15,5.4,13.9\r\n",
      "KAGGFGEPGNPAAR,GLGAGGNPSIGQK,55.556,9,4,0,4,12,1,9,6.1,13.9\r\n",
      "KAGGFGEPGNPAAR,GLGAGGNPSIGQK,55.556,9,4,0,4,12,1,9,6.1,13.9\r\n",
      "KAGGFGEPGNPAAR,RIGENTNPAK,62.500,8,3,0,5,12,2,9,7.0,13.9\r\n",
      "KAGGFGEPGNPAAR,RIGENTNPAK,62.500,8,3,0,5,12,2,9,7.0,13.9\r\n",
      "KAGGFGEPGNPAAR,IGENTNPAK,62.500,8,3,0,5,12,1,8,8.3,13.5\r\n",
      "KAGGFGEPGNPAAR,IGENTNPAK,62.500,8,3,0,5,12,1,8,8.3,13.5\r\n"
     ]
    }
   ],
   "source": [
    "! cat /Users/meganduffy/Documents/git-repos/FISH-546/analyses/062016_ETNP_129_100m_peaks_v_db.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### I dressed up the above .csv file with headers. You can see which _de novo_ peptide sequences align well with database identified peptides: ###\n",
    "\n",
    "![csv with blastp alignment](https://raw.githubusercontent.com/MeganEDuffy/FISH-546/master/images/062016_ETNP_129_100m_peaks_v_db.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
