###GO tools###

Some tools I might be able to use for my project:

- [REVIGO](http://revigo.irb.hr/)
- [UniProt](http://www.uniprot.org/help/gene_ontology)
- [DAVID](https://david.ncifcrf.gov/)

###Genome Browsers###

####Coordinate systems: looking at genome data####

What would we care about to look at in a genome broswer, and visualize:

1. where the genes are, their directionality
2. methylations
3. SNPs

####Alignment files####

SAM/BAM (BAM is binary - more compressed)
samtools has a crude genome browser embedded

bedtools:

1. intersectbed: is it flanking, do things intersect?

Integrated Genome Viewer:  

- Can run bedtools from this GUI
- Steven modified igv.sh to double the memory; he then runs this file from Terminal
- Much of his work with IGV is [here](https://sr320.github.io/student-fish546-2016/First-Aggregate/)
  - he converts a txt file from blastn output to gff using a perl script, which can go into igv
  - uses RNASeq data (we don't know intron/exon locations (annotation), so uses Bowtie2 instead of Tophat (Bowtie is part of Tophat), it's a crude approx.
  
- While I used ```curl``` to get the .xml IGV session file, I could have just used that url to open it from IGV using ```open from url```
  -This IGV session text file is basically telling IGV where to find files (urls) and how to display them (colors, etc). You can always export xml files to save these settings.
