### DNA Methylation exercise ###

In this exercise I will take a given single Olympia oyster (_O. lurida_) genome and use the [CoGe](https://genomevolution.org/coge/) comparative genomics platform to find sequences that are methylated. 

The sequences I'm analyzing can be found here: http://owl.fish.washington.edu/nightingales/O_lurida/20160203_mbdseq/zr1394_18_s6_R1.fastq.gz

After identifying highly methylated sequences, I'll BLAST them against a marine oxygen minimum zone microbial metagenome to see if there are any matches. I'm sure there will be some issues there, as there may not be much interesting homology between taxa. But the question of microbial DNA methylation is really intriguing, as DNA methylation in bacteria seems to be important as a protection against phage. From what I can tell, many bacterial DNA methyltransferases described in the literature are part of restriction-modification systems, where base methylation prevents DNA cleavage by endonucleases, thus protecting host DNA (see [Sanchez-Romero et al., 2015](http://www.sciencedirect.com.offcampus.lib.washington.edu/science/article/pii/S1369527415000399)).

Virus-host relationships are poorly understood in marine systems, but the concept of the 'viral shunt' posits that phage lysis is an important means of releasing nutrients and carbon back into seawater to fuel microbial heterotrophy and autotrophy. This process could 'prime' of the microbial loop, rather than microbial cells being soley the fuel of high trophic levels like zooplankton. 

![viral shunt](https://raw.githubusercontent.com/MeganEDuffy/FISH-546/master/tutorials/dnameth-tutorial/images/viralshunt.jpg)

